ludwig/models/llm.py

```diff
@@ -46,12 +46,16 @@ def __init__(
 
         logger.info("Loading large language model...")
         self.model = AutoModelForCausalLM.from_pretrained(self.config_obj.model_name)
+        ####SXK should we make this configurable ######
         self.curr_device = torch.device("cpu")  # model initially loaded onto cpu
         logger.info("Done.")
 
+        ####SXK why do we need this adapter and PEFT model, sorry about my ignorance here, how many types
+        ####of adapters can there be
```

Comment:
PEFT is parameter efficient fine-tuning. It basically allows you to plug and play different "adapters" that are used for fine-tuning as opposed to adjusting the weights of the base LLM model. The forward pass goes through both, but the backward pass just goes through the PEFT adapter weights.

There's many - we already support 5, and more will be added in the future.